### DISCUSSION TITLE
Show hx-indicator only if the request takes more than Xms

### DISCUSSION THREAD

**Discussion by davidjr82** [bigskysoftware/htmx#1510] [Q&A] [answered]
I have requests that are sometimes fast and other times slow.

Using hx-indicator is great when it comes to slow requests, but when the request is fast, looks jumpy (change height very fast, etc.).

Is there any way to set the htmx-request class in the indicators only if the request is taking more than a certain time? Something similar as delay:300ms, for example, but with indicators.

Thanks!

**Comment by andryyy** [comment, marked as answer]
Hi,

I think you can add `transition-delay: 500ms` to the indicator class.

**Reply by davidjr82** [reply]
You are totally right, sometimes we make it really more complex than what it is! 🗡️

**Comment by davidjr82** [comment]
I ended up using the [loading-states](https://htmx.org/extensions/loading-states/) extension, because the display property is not animatable. This extension has a `data-loading-delay` attribute which is exactly what I was looking for. Thanks!
